src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -547,6 +560,112 @@ impl EKS {
         Ok(context)
     }
 
+    /// Returns a tuple of (update_desired_node: bool, desired_nodes_count: i32).
+    fn should_update_desired_nodes(&self, event_details: EventDetails) -> Result<(bool, i32), EngineError> {
+        let future_node_group = match self.nodes_groups.is_empty() {
+            false => self.nodes_groups.first().expect("unable to get node_group"),
+            true => {
+                return Err(EngineError::new_cluster_has_no_worker_nodes(
+                    event_details,
+                    Option::Some(CommandError::new_from_safe_message(
+                        "Could not find node_group in terra context".to_string(),
+                    )),
+                ))
+            }
+        };
+
+        let region = match RusotoRegion::from_str(&self.region()) {
+            Ok(value) => value,
+            Err(error) => {
+                return Err(EngineError::new_unsupported_region(
+                    event_details,
+                    "".to_string(),
+                    CommandError::new_from_safe_message(error.to_string()),
+                ))
+            }
+        };
+        let credentials = StaticProvider::new(
+            self.cloud_provider.access_key_id(),
+            self.cloud_provider.secret_access_key(),
+            None,
+            None,
+        );
+        let client = Client::new_with(credentials, HttpClient::new().expect("unable to create new Http client"));
+        let eks_client = EksClient::new_with_client(client, region);
+
+        let node_groups = match block_on(eks_client.list_nodegroups(ListNodegroupsRequest {
+            cluster_name: self.cluster_name(),
+            ..Default::default()
+        })) {
+            Ok(res) => res.nodegroups.unwrap_or_default(),
+            Err(_) => return Ok((false, future_node_group.min_nodes)),
+        };
+
+        if node_groups.is_empty() {
+            return Err(EngineError::new_cluster_has_no_worker_nodes(
+                event_details,
+                Option::Some(CommandError::new_from_safe_message(format!(
+                    "Could not find node_groups for cluster {}",
+                    self.cluster_name()
+                ))),
+            ));
+        }
+
+        let actual_nodes_group = match block_on(eks_client.describe_nodegroup(DescribeNodegroupRequest {
```

Comment:
Don't you have a list? Today we only have one, but technically we can have several. May be having another way (filters) to distinguish them could be great